uart_emu/example/rmt_uart/main/uart_emu_main.c

```diff
@@ -0,0 +1,126 @@
+/*
+ * SPDX-FileCopyrightText: 2025 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: CC0-1.0
+ */
+#include <stdio.h>
+#include <string.h>
+#include "sdkconfig.h"
+
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+
+#include "esp_log.h"
+#include "esp_check.h"
+#include "uart_emu_rmt.h"
+
+// the memory size of each RMT channel, in words (4 bytes)
+#if SOC_RMT_SUPPORT_DMA
+#define UART_EMU_RMT_MEM_BLOCK_SYMBOLS   1024
+#else
+#define UART_EMU_RMT_MEM_BLOCK_SYMBOLS   SOC_RMT_MEM_WORDS_PER_CHANNEL
+#endif
+
+#define UART_EMU_RMT_INTR_PRIORITY 3                // use higher priority to avoid timing issues
+#define UART_EMU_RMT_TX_TRANS_QUEUE_DEPTH 10        // the max number of the pending tx transaction
+#define UART_EMU_RX_BUFFER_SIZE 128                 // the max size of the received data, in bytes
+#define UART_EMU_RMT_RX_PINGPONG_BUFFER_SIZE 256    // in rmt symbol words, the size of the ping-pong buffer
+#define UART_EMU_TX_PIN  GPIO_NUM_1
+#define UART_EMU_RX_PIN  GPIO_NUM_0
+#define UART_EMU_BAUD_RATE 115200
+
+const char *TAG = "uart_emu_rmt_example";
+
+IRAM_ATTR static bool uart_emu_rmt_tx_event_cbs(uart_emu_device_handle_t tx_unit, const uart_emu_tx_done_event_data_t *edata, void *user_ctx)
+{
+    BaseType_t high_task_wakeup = pdFALSE;
+    TaskHandle_t task = (TaskHandle_t)user_ctx;
+    vTaskNotifyGiveFromISR(task, &high_task_wakeup);
+    return high_task_wakeup == pdTRUE;
+}
+
+IRAM_ATTR static bool uart_emu_rmt_rx_event_cbs(uart_emu_device_handle_t rx_unit, const uart_emu_rx_done_event_data_t *edata, void *user_ctx)
+{
+    BaseType_t high_task_wakeup = pdFALSE;
+    QueueHandle_t receive_queue = (QueueHandle_t)user_ctx;
+    xQueueSendFromISR(receive_queue, edata, &high_task_wakeup);
+    return high_task_wakeup == pdTRUE;
+}
+
+void uart_read_task(void *pvParameters)
+{
+    uart_emu_device_handle_t uart_device = (uart_emu_device_handle_t)pvParameters;
+    QueueHandle_t receive_queue = xQueueCreate(10, sizeof(uart_emu_rx_done_event_data_t));   // queue for receive rmt rx done event
+    assert(receive_queue != NULL);
+    uart_emu_event_rx_callbacks_t uart_rx_cbs = {
+        .on_rx_trans_done = uart_emu_rmt_rx_event_cbs,
+    };
+    ESP_ERROR_CHECK(uart_emu_register_rx_event_callbacks(uart_device, &uart_rx_cbs, receive_queue));
+    uart_emu_rx_done_event_data_t rx_done_event_data;
+    rmt_symbol_word_t *receive_symbols = heap_caps_calloc(UART_EMU_RMT_RX_PINGPONG_BUFFER_SIZE, sizeof(rmt_symbol_word_t),
+                                         MALLOC_CAP_8BIT | MALLOC_CAP_INTERNAL | MALLOC_CAP_DMA);
+    assert(receive_symbols != NULL);
+
+    char read_buffer[UART_EMU_RX_BUFFER_SIZE] = {0};
+    while (1) {
+        /* Read from the UART */
+        ESP_ERROR_CHECK(uart_emu_receive(uart_device, (uint8_t *)receive_symbols, UART_EMU_RMT_RX_PINGPONG_BUFFER_SIZE * sizeof(rmt_symbol_word_t)));
+        if (xQueueReceive(receive_queue, &rx_done_event_data, portMAX_DELAY)) {
```

Comment:
how can we receive a long package? the `uart_emu_rmt_rx_event_cbs` will be called multiple times. How many times do you want the user to call the `uart_emu_receive`?